### ISSUE TITLE
ReplicaSet in scaledown delay scales up and down repeatedly

### ISSUE THREAD

**Issue by bpoland** [argoproj/argo-rollouts#1822]
## Summary 

When a Rollout has a ReplicaSet in scaledown delay that is not fully scaled up, the controller fights with itself trying to scale up/scale down pods to 100% until the scaledown delay is over.

In the case of the logs below, here were the steps I took
- Rollout is in healthy state with 10 pods, no dynamic stable scale, with a workloadRef Deployment on image A
- Start a deploy by updating workloadRef deployment to image B
- Wait for the Rollout to proceed part of the way through the steps.  Image A ReplicaSet is still at 10 pods, image B is at 5 pods.
- Update the image in the workloadRef deployment back to image A
- Rollout immediately reverts to the image A RS and becomes healthy, image B RS enters scaledown delay
- I see the image B RS scaling up from 5 to 10 pods, and then back down to 5 repeatedly
- This pod scaling up and down continues to happen until the scaledown delay is over and the image B RS scales down fully

You can see in the logs below that these 2 lines are repeated 39 times each during the 30 second scaledown delay:
```
Scaled up ReplicaSet name-5dc64db687 (revision 754) from 5 to 10
Scaled down ReplicaSet name-5dc64db687 (revision 754) from 10 to 5
```

I can reproduce this issue pretty consistently so hopefully you will be able to as well :)

## Diagnostics

What version of Argo Rollouts are you running? 1.1.1

Rollout controller logs: https://gist.github.com/bpoland/ec45808b4715f112b3d82898a1ec8db4

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

**Comment by huikang** [comment]
It sounds like the controller considers that the old replicaset should have 10 replicas and keeps bringing up the other 5 pods. 

I think the correct log order should be

```
time="2022-01-29T18:42:04Z" level=info msg="Found 5 available pods in old RS default/name-5dc64db687" namespace=default rollout=name
time="2022-01-29T18:42:04Z" level=info msg="Found 15 available pods, scaling down old RSes (minAvailable: 6, maxScaleDown: 9)" namespace=default rollout=name
time="2022-01-29T18:42:04Z" level=info msg="Set 'scale-down-deadline' annotation on 'name-5dc64db687' to 2022-01-29T18:42:34Z (30s)" namespace=default rollout=name
```
After the last line, there should be no more 
 `time="2022-01-29T18:42:04Z" level=info msg="Scaled up ReplicaSet name-5dc64db687 (revision 754) from 5 to 10" event_reason=ScalingReplicaSet namespace=default rollout=name` until the `scaledowndelay` deadline.

The following seems need to be updated:

https://github.com/argoproj/argo-rollouts/blob/0309eb6b4a2176995be1842a0f8d6bfb7f6d9943/rollout/replicaset.go#L281-L290

`desiredReplicaCount` should be 5, instead of 10 which is the rollout replicas.


Could you help verify by checking the value of `spec.replicas` of the Replicaset for image B? Thanks.

**Comment by bpoland** [comment]
Thank you for the quick fix, looking forward to a release with this included :)
